From 94a5839d74d53832674cd6eca49c980bac27fc04 Mon Sep 17 00:00:00 2001
From: mfekari <mfekari@rim.com>
Date: Mon, 10 Dec 2012 08:51:36 -0500
Subject: [PATCH 057/246] Update defaultNumberingSystem value for some indic
 and slovak countries/regions

Qt gets the information about locale digits from the <defaultNumberingSystem>
entries in the CLDR data xm files. Current CLDR data in Qt4.8 is 2.0, which
specifies native digits for most of Indic locales and arabic digits for slovak locale.

This doesn't concern Qt5 that uses the most recent version of CLDR (22.1) which specify Latin digits for the locales in question.

This patch updates the defaultNumberingSystem (to latin) and provides an autotest to compare the returned vs expected digits for the following langagues/regions:

- Gujarati, Hindi, Kannada, Konkani, Malayalam, Marathi, Nepali, Punjabi, Tamil, Telugu (to use latin digits instead of native ones)
- Slovak (to use latin digits instead of arabic ones)

Change-Id: I04364cdb80292f7705b1321377623298a01a9c5b
Reviewed-by: Vladimir Minenko <vminenko@rim.com>
Reviewed-by: Konstantin Ritt <ritt.ks@gmail.com>
---
 src/corelib/tools/qlocale_data_p.h | 38 ++++++++++++++++++------------------
 tests/auto/qlocale/tst_qlocale.cpp | 40 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 59 insertions(+), 19 deletions(-)

diff --git a/src/corelib/tools/qlocale_data_p.h b/src/corelib/tools/qlocale_data_p.h
index 6c848d0..07ebd36 100644
--- a/src/corelib/tools/qlocale_data_p.h
+++ b/src/corelib/tools/qlocale_data_p.h
@@ -75,7 +75,7 @@ static const int ImperialMeasurementSystemsCount =
 // GENERATED PART STARTS HERE
 
 /*
-    This part of the file was generated on 2012-11-02 from the
+    This part of the file was generated on 2012-12-11 from the
     Common Locale Data Repository v2.0
 
     http://www.unicode.org/cldr/
@@ -464,7 +464,7 @@ static const QLocalePrivate locale_data[] = {
     {     43,     0,    85,    44,    46,    44,    37,    48,    45,    43,   101,   171,   187,  8216,  8217, 0,6 , 0,6 , 178,9 , 178,9 , 279,6 , 10,17 , 18,7 , 25,12 , 5361,50 , 5411,115 , 5526,24 , 5430,50 , 5480,115 , 5595,24 , 3402,28 , 3430,55 , 3485,14 , 3499,28 , 3430,55 , 3485,14 , 112,4 , 102,4 , {69,85,82},  112,1 , 3747,19 , 25,5 , 4,0 , 1544,8 , 1552,6 ,     2,     1,     1,     6,     7 }, // Greek/AnyScript/Greece
     {     43,     0,    56,    44,    46,    44,    37,    48,    45,    43,   101,   171,   187,  8216,  8217, 0,6 , 0,6 , 178,9 , 178,9 , 279,6 , 10,17 , 18,7 , 25,12 , 5361,50 , 5411,115 , 5526,24 , 5430,50 , 5480,115 , 5595,24 , 3402,28 , 3430,55 , 3485,14 , 3499,28 , 3430,55 , 3485,14 , 112,4 , 102,4 , {69,85,82},  112,1 , 3747,19 , 4,4 , 4,0 , 1544,8 , 1558,6 ,     2,     1,     1,     6,     7 }, // Greek/AnyScript/Cyprus
     {     44,     0,    86,    44,    46,    59,    37,    48,  8722,    43,   101,   187,   171,  8250,  8249, 187,11 , 0,6 , 0,6 , 187,11 , 72,10 , 82,17 , 18,7 , 25,12 , 3581,48 , 5550,96 , 134,24 , 5619,48 , 5667,96 , 320,24 , 3527,28 , 3555,98 , 3653,14 , 3527,28 , 3555,98 , 3653,14 , 0,2 , 0,2 , {68,75,75},  131,2 , 3766,24 , 4,4 , 36,5 , 1564,11 , 1575,16 ,     2,     1,     1,     6,     7 }, // Greenlandic/AnyScript/Greenland
-    {     46,     0,   100,  1643,  1644,  1563,  1642,  2790,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 198,9 , 198,9 , 609,7 , 203,18 , 345,8 , 353,13 , 5646,67 , 5713,87 , 5800,31 , 5763,67 , 5830,87 , 5917,31 , 3667,32 , 3699,53 , 3752,19 , 3667,32 , 3699,53 , 3752,19 , 116,14 , 106,14 , {73,78,82},  166,2 , 3790,20 , 8,5 , 4,0 , 1591,7 , 1598,4 ,     2,     1,     7,     7,     7 }, // Gujarati/AnyScript/India
+    {     46,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 198,9 , 198,9 , 609,7 , 203,18 , 345,8 , 353,13 , 5646,67 , 5713,87 , 5800,31 , 5763,67 , 5830,87 , 5917,31 , 3667,32 , 3699,53 , 3752,19 , 3667,32 , 3699,53 , 3752,19 , 116,14 , 106,14 , {73,78,82},  166,2 , 3790,20 , 8,5 , 4,0 , 1591,7 , 1598,4 ,     2,     1,     7,     7,     7 }, // Gujarati/AnyScript/India
     {     47,     0,    83,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8220, 0,6 , 0,6 , 0,6 , 0,6 , 279,6 , 203,18 , 37,5 , 8,10 , 5831,48 , 5879,85 , 5964,24 , 5948,48 , 5996,85 , 6081,24 , 3771,21 , 3792,52 , 3844,14 , 3771,21 , 3792,52 , 3844,14 , 0,2 , 0,2 , {71,72,83},  0,0 , 0,7 , 8,5 , 4,0 , 1602,5 , 1607,4 ,     2,     1,     1,     6,     7 }, // Hausa/AnyScript/Ghana
     {     47,     0,   156,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8220, 0,6 , 0,6 , 0,6 , 0,6 , 279,6 , 203,18 , 37,5 , 8,10 , 5831,48 , 5879,85 , 5964,24 , 5948,48 , 5996,85 , 6081,24 , 3771,21 , 3792,52 , 3844,14 , 3771,21 , 3792,52 , 3844,14 , 0,2 , 0,2 , {88,79,70},  168,3 , 3810,36 , 8,5 , 4,0 , 1602,5 , 1611,5 ,     0,     0,     1,     6,     7 }, // Hausa/AnyScript/Niger
     {     47,     0,   157,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8220, 0,6 , 0,6 , 0,6 , 0,6 , 279,6 , 203,18 , 37,5 , 8,10 , 5831,48 , 5879,85 , 5964,24 , 5948,48 , 5996,85 , 6081,24 , 3771,21 , 3792,52 , 3844,14 , 3771,21 , 3792,52 , 3844,14 , 0,2 , 0,2 , {78,71,78},  171,1 , 3846,12 , 8,5 , 4,0 , 1602,5 , 1616,8 ,     2,     1,     1,     6,     7 }, // Hausa/AnyScript/Nigeria
@@ -472,7 +472,7 @@ static const QLocalePrivate locale_data[] = {
     {     47,     7,   156,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8220, 0,6 , 0,6 , 0,6 , 0,6 , 279,6 , 203,18 , 37,5 , 8,10 , 5831,48 , 5879,85 , 5964,24 , 5948,48 , 5996,85 , 6081,24 , 3771,21 , 3792,52 , 3844,14 , 3771,21 , 3792,52 , 3844,14 , 0,2 , 0,2 , {88,79,70},  168,3 , 3810,36 , 8,5 , 4,0 , 1602,5 , 1611,5 ,     0,     0,     1,     6,     7 }, // Hausa/Latin/Niger
     {     47,     7,   157,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8220, 0,6 , 0,6 , 0,6 , 0,6 , 279,6 , 203,18 , 37,5 , 8,10 , 5831,48 , 5879,85 , 5964,24 , 5948,48 , 5996,85 , 6081,24 , 3771,21 , 3792,52 , 3844,14 , 3771,21 , 3792,52 , 3844,14 , 0,2 , 0,2 , {78,71,78},  171,1 , 3846,12 , 8,5 , 4,0 , 1602,5 , 1616,8 ,     2,     1,     1,     6,     7 }, // Hausa/Latin/Nigeria
     {     48,     0,   105,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,    34,    34, 0,6 , 0,6 , 207,7 , 207,7 , 27,8 , 616,18 , 37,5 , 8,10 , 5988,58 , 6046,72 , 1381,27 , 6105,48 , 6153,72 , 134,27 , 3858,46 , 3904,65 , 3969,14 , 3858,46 , 3904,65 , 3969,14 , 130,6 , 120,5 , {73,76,83},  172,1 , 3858,21 , 25,5 , 4,0 , 1624,5 , 1629,5 ,     2,     1,     7,     5,     6 }, // Hebrew/AnyScript/Israel
-    {     49,     0,   100,    46,    44,    59,    37,  2406,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 214,9 , 223,8 , 634,6 , 10,17 , 18,7 , 25,12 , 6118,75 , 6118,75 , 6193,30 , 6225,75 , 6225,75 , 6300,30 , 3983,38 , 4021,57 , 4078,19 , 3983,38 , 4021,57 , 4078,19 , 136,9 , 125,7 , {73,78,82},  173,3 , 3879,19 , 8,5 , 4,0 , 1634,6 , 1640,4 ,     2,     1,     7,     7,     7 }, // Hindi/AnyScript/India
+    {     49,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 214,9 , 223,8 , 634,6 , 10,17 , 18,7 , 25,12 , 6118,75 , 6118,75 , 6193,30 , 6225,75 , 6225,75 , 6300,30 , 3983,38 , 4021,57 , 4078,19 , 3983,38 , 4021,57 , 4078,19 , 136,9 , 125,7 , {73,78,82},  173,3 , 3879,19 , 8,5 , 4,0 , 1634,6 , 1640,4 ,     2,     1,     7,     7,     7 }, // Hindi/AnyScript/India
     {     50,     0,    98,    44,   160,    59,    37,    48,    45,    43,   101,  8222,  8221,  8222,  8221, 0,6 , 0,6 , 231,8 , 231,8 , 640,11 , 651,19 , 165,4 , 195,9 , 6223,64 , 6287,98 , 6385,25 , 6330,64 , 6394,98 , 6492,25 , 4097,19 , 4116,52 , 4168,17 , 4097,19 , 4116,52 , 4168,17 , 145,3 , 132,3 , {72,85,70},  176,2 , 3898,20 , 25,5 , 4,0 , 1644,6 , 1650,12 ,     0,     0,     1,     6,     7 }, // Hungarian/AnyScript/Hungary
     {     51,     0,    99,    44,    46,    59,    37,    48,  8722,    43,   101,  8222,  8220,  8218,  8216, 0,6 , 0,6 , 111,8 , 111,8 , 582,8 , 498,18 , 37,5 , 8,10 , 6410,48 , 6458,82 , 6540,24 , 6517,48 , 6565,82 , 6647,24 , 4185,28 , 4213,81 , 4294,14 , 4185,28 , 4213,81 , 4308,14 , 148,4 , 135,4 , {73,83,75},  131,2 , 3918,48 , 25,5 , 4,0 , 1662,8 , 1670,6 ,     0,     0,     1,     6,     7 }, // Icelandic/AnyScript/Iceland
     {     52,     0,   101,    44,    46,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 239,10 , 249,9 , 27,8 , 123,18 , 37,5 , 195,9 , 6564,48 , 6612,87 , 134,24 , 6671,48 , 6719,87 , 320,24 , 4322,28 , 4350,43 , 4393,14 , 4322,28 , 4350,43 , 4393,14 , 152,4 , 139,5 , {73,68,82},  178,2 , 3966,23 , 4,4 , 4,0 , 1676,16 , 1692,9 ,     0,     0,     7,     6,     7 }, // Indonesian/AnyScript/Indonesia
@@ -480,7 +480,7 @@ static const QLocalePrivate locale_data[] = {
     {     58,     0,   106,    44,    46,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 258,8 , 154,7 , 27,8 , 99,16 , 37,5 , 8,10 , 6732,48 , 6780,94 , 6874,24 , 6839,48 , 6887,94 , 6981,24 , 4439,28 , 4467,57 , 4524,14 , 4439,28 , 4538,57 , 4524,14 , 156,2 , 144,2 , {69,85,82},  112,1 , 1825,19 , 8,5 , 4,0 , 1712,8 , 1720,6 ,     2,     1,     1,     6,     7 }, // Italian/AnyScript/Italy
     {     58,     0,   206,    46,    39,    59,    37,    48,    45,    43,   101,   171,   187,  8249,  8250, 0,6 , 0,6 , 258,8 , 154,7 , 332,8 , 10,17 , 37,5 , 331,14 , 6732,48 , 6780,94 , 6874,24 , 6839,48 , 6887,94 , 6981,24 , 4439,28 , 4467,57 , 4524,14 , 4439,28 , 4538,57 , 4524,14 , 156,2 , 144,2 , {67,72,70},  163,3 , 3989,53 , 8,5 , 48,5 , 1712,8 , 1726,8 ,     2,     5,     1,     6,     7 }, // Italian/AnyScript/Switzerland
     {     59,     0,   108,    46,    44,    59,    37,    48,    45,    43,   101, 12300, 12301, 12302, 12303, 94,5 , 94,5 , 94,5 , 94,5 , 221,8 , 429,13 , 165,4 , 366,10 , 3254,39 , 3254,39 , 1381,27 , 3278,39 , 3278,39 , 134,27 , 4595,14 , 4609,28 , 4595,14 , 4595,14 , 4609,28 , 4595,14 , 158,2 , 146,2 , {74,80,89},  123,1 , 4042,10 , 4,4 , 4,0 , 1734,3 , 1737,2 ,     0,     0,     7,     6,     7 }, // Japanese/AnyScript/Japan
-    {     61,     0,   100,  1643,  1644,  1563,  1642,  3302,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 266,12 , 278,11 , 634,6 , 99,16 , 345,8 , 353,13 , 6898,86 , 6898,86 , 6984,31 , 7005,86 , 7005,86 , 7091,31 , 4637,28 , 4665,53 , 4718,19 , 4637,28 , 4665,53 , 4718,19 , 0,2 , 0,2 , {73,78,82},  114,1 , 4052,20 , 8,5 , 4,0 , 1739,5 , 1744,4 ,     2,     1,     7,     7,     7 }, // Kannada/AnyScript/India
+    {     61,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 266,12 , 278,11 , 634,6 , 99,16 , 345,8 , 353,13 , 6898,86 , 6898,86 , 6984,31 , 7005,86 , 7005,86 , 7091,31 , 4637,28 , 4665,53 , 4718,19 , 4637,28 , 4665,53 , 4718,19 , 0,2 , 0,2 , {73,78,82},  114,1 , 4052,20 , 8,5 , 4,0 , 1739,5 , 1744,4 ,     2,     1,     7,     7,     7 }, // Kannada/AnyScript/India
     {     63,     0,   110,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 332,8 , 670,22 , 37,5 , 8,10 , 7015,61 , 7076,83 , 1381,27 , 7122,61 , 7183,83 , 134,27 , 4737,28 , 4765,56 , 708,14 , 4821,28 , 4849,54 , 708,14 , 0,2 , 0,2 , {75,90,84},  180,4 , 4072,24 , 25,5 , 4,0 , 1748,10 , 1758,9 ,     2,     1,     1,     6,     7 }, // Kazakh/AnyScript/Kazakhstan
     {     63,     2,   110,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 332,8 , 670,22 , 37,5 , 8,10 , 7015,61 , 7076,83 , 1381,27 , 7122,61 , 7183,83 , 134,27 , 4737,28 , 4765,56 , 708,14 , 4821,28 , 4849,54 , 708,14 , 0,2 , 0,2 , {75,90,84},  180,4 , 4072,24 , 25,5 , 4,0 , 1748,10 , 1758,9 ,     2,     1,     1,     6,     7 }, // Kazakh/Cyrillic/Kazakhstan
     {     64,     0,   179,    44,    46,    59,    37,    48,    45,    43,   101,  8216,  8217,   171,   187, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 7159,60 , 7219,101 , 1381,27 , 7266,60 , 7326,101 , 134,27 , 4903,35 , 4938,84 , 708,14 , 4903,35 , 4938,84 , 708,14 , 0,2 , 0,2 , {82,87,70},  184,2 , 0,7 , 8,5 , 4,0 , 1767,11 , 1340,6 ,     0,     0,     1,     6,     7 }, // Kinyarwanda/AnyScript/Rwanda
@@ -488,8 +488,8 @@ static const QLocalePrivate locale_data[] = {
     {     66,     0,   114,    46,    44,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 289,7 , 289,7 , 692,9 , 701,16 , 376,7 , 383,13 , 7320,39 , 7320,39 , 7320,39 , 7427,39 , 7427,39 , 7427,39 , 5022,14 , 5036,28 , 5022,14 , 5022,14 , 5036,28 , 5022,14 , 160,2 , 148,2 , {75,82,87},  189,1 , 4096,13 , 4,4 , 4,0 , 1794,3 , 1797,4 ,     0,     0,     7,     6,     7 }, // Korean/AnyScript/RepublicOfKorea
     {     67,     0,   102,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7359,105 , 7359,105 , 7464,26 , 7466,105 , 7466,105 , 7571,26 , 5064,58 , 5064,58 , 5122,14 , 5064,58 , 5064,58 , 5122,14 , 0,2 , 0,2 , {73,82,82},  0,0 , 4109,19 , 8,5 , 4,0 , 1801,5 , 1806,5 ,     0,     0,     6,     4,     5 }, // Kurdish/AnyScript/Iran
     {     67,     0,   103,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7359,105 , 7359,105 , 7464,26 , 7466,105 , 7466,105 , 7571,26 , 5064,58 , 5064,58 , 5122,14 , 5064,58 , 5064,58 , 5122,14 , 0,2 , 0,2 , {73,81,68},  0,0 , 4128,20 , 8,5 , 4,0 , 1801,5 , 1811,5 ,     0,     0,     6,     5,     6 }, // Kurdish/AnyScript/Iraq
-    {     67,     0,   207,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7490,41 , 7531,51 , 7582,27 , 7597,41 , 7638,51 , 7689,27 , 5136,20 , 5156,39 , 5195,14 , 5136,20 , 5156,39 , 5195,14 , 0,2 , 0,2 , {83,89,80},  0,0 , 0,7 , 8,5 , 4,0 , 1816,5 , 0,0 ,     0,     0,     6,     5,     6 }, // Kurdish/AnyScript/SyrianArabRepublic
-    {     67,     0,   217,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7490,41 , 7531,51 , 7582,27 , 7597,41 , 7638,51 , 7689,27 , 5136,20 , 5156,39 , 5195,14 , 5136,20 , 5156,39 , 5195,14 , 0,2 , 0,2 , {84,82,89},  190,2 , 0,7 , 8,5 , 4,0 , 1816,5 , 1821,7 ,     2,     1,     1,     6,     7 }, // Kurdish/AnyScript/Turkey
+    {     67,     0,   207,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7490,41 , 7531,51 , 7582,27 , 7597,41 , 7638,51 , 7689,27 , 5136,20 , 5156,39 , 5195,14 , 5136,20 , 5156,39 , 5195,14 , 0,2 , 0,2 , {83,89,80},  0,0 , 0,7 , 8,5 , 4,0 , 1816,5 , 0,0 ,     0,     0,     6,     5,     6 }, // Kurdish/AnyScript/SyrianArabRepublic
+    {     67,     0,   217,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7490,41 , 7531,51 , 7582,27 , 7597,41 , 7638,51 , 7689,27 , 5136,20 , 5156,39 , 5195,14 , 5136,20 , 5156,39 , 5195,14 , 0,2 , 0,2 , {84,82,89},  190,2 , 0,7 , 8,5 , 4,0 , 1816,5 , 1821,7 ,     2,     1,     1,     6,     7 }, // Kurdish/AnyScript/Turkey
     {     67,     1,   102,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7359,105 , 7359,105 , 7464,26 , 7466,105 , 7466,105 , 7571,26 , 5064,58 , 5064,58 , 5122,14 , 5064,58 , 5064,58 , 5122,14 , 0,2 , 0,2 , {73,82,82},  0,0 , 4109,19 , 8,5 , 4,0 , 1801,5 , 1806,5 ,     0,     0,     6,     4,     5 }, // Kurdish/Arabic/Iran
     {     67,     1,   103,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7359,105 , 7359,105 , 7464,26 , 7466,105 , 7466,105 , 7571,26 , 5064,58 , 5064,58 , 5122,14 , 5064,58 , 5064,58 , 5122,14 , 0,2 , 0,2 , {73,81,68},  0,0 , 4128,20 , 8,5 , 4,0 , 1801,5 , 1811,5 ,     0,     0,     6,     5,     6 }, // Kurdish/Arabic/Iraq
     {     67,     7,   207,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 7490,41 , 7531,51 , 7582,27 , 7597,41 , 7638,51 , 7689,27 , 5136,20 , 5156,39 , 5195,14 , 5136,20 , 5156,39 , 5195,14 , 0,2 , 0,2 , {83,89,80},  0,0 , 0,7 , 8,5 , 4,0 , 1816,5 , 0,0 ,     0,     0,     6,     5,     6 }, // Kurdish/Latin/SyrianArabRepublic
@@ -504,15 +504,15 @@ static const QLocalePrivate locale_data[] = {
     {     75,     0,   128,    46,    44,    59,    37,    48,    45,    43,   101,   171,   187,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 364,8 , 99,16 , 37,5 , 8,10 , 8715,48 , 8763,92 , 134,24 , 8821,48 , 8869,92 , 320,24 , 5874,34 , 5908,60 , 5968,14 , 5874,34 , 5908,60 , 5968,14 , 0,2 , 0,2 , {77,71,65},  0,0 , 4366,13 , 4,4 , 4,0 , 1938,8 , 1946,12 ,     0,     0,     1,     6,     7 }, // Malagasy/AnyScript/Madagascar
     {     76,     0,   130,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 239,10 , 249,9 , 547,7 , 10,17 , 18,7 , 25,12 , 8855,49 , 8904,82 , 8986,24 , 8961,49 , 9010,82 , 9092,24 , 5982,28 , 6010,43 , 6053,14 , 5982,28 , 6010,43 , 6053,14 , 208,2 , 186,3 , {77,89,82},  201,2 , 4379,23 , 4,4 , 13,6 , 1958,13 , 1971,8 ,     2,     1,     1,     6,     7 }, // Malay/AnyScript/Malaysia
     {     76,     0,    32,    44,    46,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 239,10 , 249,9 , 547,7 , 560,12 , 18,7 , 25,12 , 8855,49 , 8904,82 , 8986,24 , 8961,49 , 9010,82 , 9092,24 , 5982,28 , 6010,43 , 6053,14 , 5982,28 , 6010,43 , 6053,14 , 208,2 , 186,3 , {66,78,68},  12,1 , 4402,19 , 8,5 , 4,0 , 1958,13 , 1979,6 ,     2,     1,     1,     6,     7 }, // Malay/AnyScript/BruneiDarussalam
-    {     77,     0,   100,    46,    44,    59,    37,  3430,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 312,13 , 325,12 , 27,8 , 795,18 , 18,7 , 25,12 , 9010,66 , 9076,101 , 9177,31 , 9116,66 , 9182,101 , 9283,31 , 6067,47 , 6114,70 , 6184,22 , 6067,47 , 6114,70 , 6184,22 , 210,6 , 189,10 , {73,78,82},  203,2 , 4421,46 , 0,4 , 4,0 , 1985,6 , 1991,6 ,     2,     1,     7,     7,     7 }, // Malayalam/AnyScript/India
+    {     77,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 312,13 , 325,12 , 27,8 , 795,18 , 18,7 , 25,12 , 9010,66 , 9076,101 , 9177,31 , 9116,66 , 9182,101 , 9283,31 , 6067,47 , 6114,70 , 6184,22 , 6067,47 , 6114,70 , 6184,22 , 210,6 , 189,10 , {73,78,82},  203,2 , 4421,46 , 0,4 , 4,0 , 1985,6 , 1991,6 ,     2,     1,     7,     7,     7 }, // Malayalam/AnyScript/India
     {     78,     0,   133,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 813,23 , 37,5 , 8,10 , 9208,48 , 9256,86 , 9342,24 , 9314,48 , 9362,86 , 9448,24 , 6206,28 , 6234,63 , 6297,14 , 6206,28 , 6234,63 , 6297,14 , 216,2 , 199,2 , {69,85,82},  112,1 , 4467,11 , 4,4 , 4,0 , 1997,5 , 774,5 ,     2,     1,     7,     6,     7 }, // Maltese/AnyScript/Malta
-    {     80,     0,   100,  1643,  1644,  1563,  1642,  2406,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 337,9 , 337,9 , 634,6 , 99,16 , 417,7 , 424,12 , 9366,69 , 9435,86 , 9521,32 , 9472,69 , 9541,86 , 9627,32 , 6311,32 , 6343,53 , 4078,19 , 6311,32 , 6343,53 , 4078,19 , 218,5 , 201,5 , {73,78,82},  205,2 , 4478,19 , 8,5 , 4,0 , 2002,5 , 1640,4 ,     2,     1,     7,     7,     7 }, // Marathi/AnyScript/India
+    {     80,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 337,9 , 337,9 , 634,6 , 99,16 , 417,7 , 424,12 , 9366,69 , 9435,86 , 9521,32 , 9472,69 , 9541,86 , 9627,32 , 6311,32 , 6343,53 , 4078,19 , 6311,32 , 6343,53 , 4078,19 , 218,5 , 201,5 , {73,78,82},  205,2 , 4478,19 , 8,5 , 4,0 , 2002,5 , 1640,4 ,     2,     1,     7,     7,     7 }, // Marathi/AnyScript/India
     {     82,     0,   143,    44,   160,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9553,48 , 9601,66 , 1381,27 , 9659,48 , 9707,66 , 134,27 , 6396,21 , 6417,43 , 708,14 , 6396,21 , 6417,43 , 708,14 , 0,2 , 0,2 , {77,78,84},  207,1 , 0,7 , 8,5 , 4,0 , 2007,6 , 2013,10 ,     0,     0,     1,     6,     7 }, // Mongolian/AnyScript/Mongolia
     {     82,     0,    44,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 1381,27 , 1381,27 , 1381,27 , 134,27 , 134,27 , 134,27 , 708,14 , 708,14 , 708,14 , 708,14 , 708,14 , 708,14 , 0,2 , 0,2 , {67,78,89},  208,3 , 0,7 , 8,5 , 4,0 , 0,0 , 0,0 ,     2,     1,     7,     6,     7 }, // Mongolian/AnyScript/China
     {     82,     2,   143,    44,   160,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9553,48 , 9601,66 , 1381,27 , 9659,48 , 9707,66 , 134,27 , 6396,21 , 6417,43 , 708,14 , 6396,21 , 6417,43 , 708,14 , 0,2 , 0,2 , {77,78,84},  207,1 , 0,7 , 8,5 , 4,0 , 2007,6 , 2013,10 ,     0,     0,     1,     6,     7 }, // Mongolian/Cyrillic/Mongolia
     {     82,     8,    44,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 1381,27 , 1381,27 , 1381,27 , 134,27 , 134,27 , 134,27 , 708,14 , 708,14 , 708,14 , 708,14 , 708,14 , 708,14 , 0,2 , 0,2 , {67,78,89},  208,3 , 0,7 , 8,5 , 4,0 , 0,0 , 0,0 ,     2,     1,     7,     6,     7 }, // Mongolian/Mongolian/China
-    {     84,     0,   150,    46,    44,    59,  1642,  2406,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9667,56 , 9723,85 , 9808,27 , 9773,56 , 9829,85 , 9914,27 , 6460,33 , 6493,54 , 6547,14 , 6460,33 , 6493,54 , 6547,14 , 223,14 , 206,14 , {78,80,82},  211,4 , 0,7 , 8,5 , 4,0 , 2023,6 , 2029,5 ,     2,     1,     7,     6,     7 }, // Nepali/AnyScript/Nepal
-    {     84,     0,   100,    46,    44,    59,  1642,  2406,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9667,56 , 9835,80 , 9808,27 , 9773,56 , 9941,80 , 9914,27 , 6460,33 , 6561,54 , 6547,14 , 6460,33 , 6561,54 , 6547,14 , 136,9 , 125,7 , {73,78,82},  114,1 , 4497,49 , 8,5 , 4,0 , 2023,6 , 1640,4 ,     2,     1,     7,     7,     7 }, // Nepali/AnyScript/India
+    {     84,     0,   150,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9667,56 , 9723,85 , 9808,27 , 9773,56 , 9829,85 , 9914,27 , 6460,33 , 6493,54 , 6547,14 , 6460,33 , 6493,54 , 6547,14 , 223,14 , 206,14 , {78,80,82},  211,4 , 0,7 , 8,5 , 4,0 , 2023,6 , 2029,5 ,     2,     1,     7,     6,     7 }, // Nepali/AnyScript/Nepal
+    {     84,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9667,56 , 9835,80 , 9808,27 , 9773,56 , 9941,80 , 9914,27 , 6460,33 , 6561,54 , 6547,14 , 6460,33 , 6561,54 , 6547,14 , 136,9 , 125,7 , {73,78,82},  114,1 , 4497,49 , 8,5 , 4,0 , 2023,6 , 1640,4 ,     2,     1,     7,     7,     7 }, // Nepali/AnyScript/India
     {     85,     0,   161,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 111,8 , 111,8 , 332,8 , 836,17 , 37,5 , 436,16 , 4293,48 , 9915,83 , 134,24 , 10021,59 , 10080,83 , 320,24 , 6615,28 , 2293,51 , 2344,14 , 6643,35 , 2293,51 , 2344,14 , 0,2 , 0,2 , {78,79,75},  131,2 , 4546,44 , 8,5 , 4,0 , 2034,12 , 2046,5 ,     2,     1,     1,     6,     7 }, // Norwegian/AnyScript/Norway
     {     86,     0,    74,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 9998,83 , 9998,83 , 1381,27 , 10163,83 , 10163,83 , 134,27 , 6678,57 , 6678,57 , 708,14 , 6678,57 , 6678,57 , 708,14 , 0,2 , 0,2 , {69,85,82},  112,1 , 1510,11 , 8,5 , 4,0 , 2051,7 , 2058,6 ,     2,     1,     1,     6,     7 }, // Occitan/AnyScript/France
     {     87,     0,   100,    46,    44,  1563,  1642,  2918,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 634,6 , 10,17 , 18,7 , 25,12 , 10081,89 , 10081,89 , 10170,32 , 10246,89 , 10246,89 , 10335,32 , 6735,33 , 6768,54 , 6822,18 , 6735,33 , 6768,54 , 6822,18 , 237,2 , 220,2 , {73,78,82},  114,1 , 4590,11 , 8,5 , 4,0 , 2064,5 , 2069,4 ,     2,     1,     7,     7,     7 }, // Oriya/AnyScript/India
@@ -526,10 +526,10 @@ static const QLocalePrivate locale_data[] = {
     {     91,     0,    92,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 154,7 , 154,7 , 27,8 , 883,27 , 37,5 , 195,9 , 10557,48 , 10605,89 , 134,24 , 10836,48 , 10884,89 , 320,24 , 7010,28 , 7038,79 , 7117,14 , 7010,28 , 7038,79 , 7117,14 , 61,4 , 59,4 , {88,79,70},  168,3 , 4859,62 , 25,5 , 4,0 , 2136,9 , 2176,12 ,     0,     0,     1,     6,     7 }, // Portuguese/AnyScript/GuineaBissau
     {     91,     0,   146,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 154,7 , 154,7 , 27,8 , 883,27 , 37,5 , 195,9 , 10557,48 , 10605,89 , 134,24 , 10836,48 , 10884,89 , 320,24 , 7010,28 , 7038,79 , 7117,14 , 7010,28 , 7038,79 , 7117,14 , 61,4 , 59,4 , {77,90,78},  223,3 , 4921,72 , 25,5 , 4,0 , 2136,9 , 2188,10 ,     2,     1,     7,     6,     7 }, // Portuguese/AnyScript/Mozambique
     {     91,     0,   185,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 154,7 , 154,7 , 27,8 , 883,27 , 37,5 , 195,9 , 10557,48 , 10605,89 , 134,24 , 10836,48 , 10884,89 , 320,24 , 7010,28 , 7038,79 , 7117,14 , 7010,28 , 7038,79 , 7117,14 , 61,4 , 59,4 , {83,84,68},  226,2 , 4993,92 , 25,5 , 4,0 , 2136,9 , 2198,19 ,     0,     0,     1,     6,     7 }, // Portuguese/AnyScript/SaoTomeAndPrincipe
-    {     92,     0,   100,  1643,  1644,  1563,  1642,  2662,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10831,68 , 10831,68 , 10899,27 , 11110,68 , 11110,68 , 11178,27 , 7210,38 , 7248,55 , 7303,23 , 7210,38 , 7248,55 , 7303,23 , 252,11 , 234,11 , {73,78,82},  228,3 , 5085,12 , 8,5 , 4,0 , 2217,6 , 2223,4 ,     2,     1,     7,     7,     7 }, // Punjabi/AnyScript/India
-    {     92,     0,   163,  1643,  1644,  1563,  1642,  1776,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10926,67 , 10926,67 , 1381,27 , 11205,67 , 11205,67 , 134,27 , 7326,37 , 7326,37 , 708,14 , 7326,37 , 7326,37 , 708,14 , 0,2 , 0,2 , {80,75,82},  231,1 , 5097,13 , 8,5 , 4,0 , 2227,5 , 2232,6 ,     0,     0,     7,     6,     7 }, // Punjabi/AnyScript/Pakistan
+    {     92,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10831,68 , 10831,68 , 10899,27 , 11110,68 , 11110,68 , 11178,27 , 7210,38 , 7248,55 , 7303,23 , 7210,38 , 7248,55 , 7303,23 , 252,11 , 234,11 , {73,78,82},  228,3 , 5085,12 , 8,5 , 4,0 , 2217,6 , 2223,4 ,     2,     1,     7,     7,     7 }, // Punjabi/AnyScript/India
+    {     92,     0,   163,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10926,67 , 10926,67 , 1381,27 , 11205,67 , 11205,67 , 134,27 , 7326,37 , 7326,37 , 708,14 , 7326,37 , 7326,37 , 708,14 , 0,2 , 0,2 , {80,75,82},  231,1 , 5097,13 , 8,5 , 4,0 , 2227,5 , 2232,6 ,     0,     0,     7,     6,     7 }, // Punjabi/AnyScript/Pakistan
     {     92,     1,   163,  1643,  1644,  1563,  1642,  1776,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10926,67 , 10926,67 , 1381,27 , 11205,67 , 11205,67 , 134,27 , 7326,37 , 7326,37 , 708,14 , 7326,37 , 7326,37 , 708,14 , 0,2 , 0,2 , {80,75,82},  231,1 , 5097,13 , 8,5 , 4,0 , 2227,5 , 2232,6 ,     0,     0,     7,     6,     7 }, // Punjabi/Arabic/Pakistan
-    {     92,     4,   100,  1643,  1644,  1563,  1642,  2662,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10831,68 , 10831,68 , 10899,27 , 11110,68 , 11110,68 , 11178,27 , 7210,38 , 7248,55 , 7303,23 , 7210,38 , 7248,55 , 7303,23 , 252,11 , 234,11 , {73,78,82},  228,3 , 5085,12 , 8,5 , 4,0 , 2217,6 , 2223,4 ,     2,     1,     7,     7,     7 }, // Punjabi/Gurmukhi/India
+    {     92,     4,   100,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 123,18 , 18,7 , 25,12 , 10831,68 , 10831,68 , 10899,27 , 11110,68 , 11110,68 , 11178,27 , 7210,38 , 7248,55 , 7303,23 , 7210,38 , 7248,55 , 7303,23 , 252,11 , 234,11 , {73,78,82},  228,3 , 5085,12 , 8,5 , 4,0 , 2217,6 , 2223,4 ,     2,     1,     7,     7,     7 }, // Punjabi/Gurmukhi/India
     {     94,     0,   206,    46,  8217,    59,    37,    48,  8722,    43,   101,   171,   187,  8249,  8250, 0,6 , 0,6 , 0,6 , 0,6 , 332,8 , 498,18 , 37,5 , 8,10 , 10993,67 , 11060,92 , 11152,24 , 11272,67 , 11339,92 , 11431,24 , 7363,23 , 7386,56 , 7442,14 , 7363,23 , 7386,56 , 7442,14 , 237,2 , 245,2 , {67,72,70},  0,0 , 5110,20 , 25,5 , 4,0 , 2238,9 , 2247,6 ,     2,     5,     1,     6,     7 }, // RhaetoRomance/AnyScript/Switzerland
     {     95,     0,   141,    44,    46,    59,    37,    48,    45,    43,   101,  8222,  8221,   171,   187, 0,6 , 0,6 , 359,8 , 359,8 , 873,10 , 10,17 , 37,5 , 8,10 , 11176,60 , 11236,98 , 11334,24 , 11455,60 , 11515,98 , 11613,24 , 7456,21 , 7477,48 , 3070,14 , 7456,21 , 7477,48 , 3070,14 , 0,2 , 0,2 , {77,68,76},  232,3 , 5130,69 , 25,5 , 4,0 , 2253,6 , 2259,17 ,     2,     1,     1,     6,     7 }, // Romanian/AnyScript/Moldova
     {     95,     0,   177,    44,    46,    59,    37,    48,    45,    43,   101,  8222,  8221,   171,   187, 0,6 , 0,6 , 359,8 , 359,8 , 873,10 , 10,17 , 37,5 , 8,10 , 11176,60 , 11236,98 , 11334,24 , 11455,60 , 11515,98 , 11613,24 , 7456,21 , 7477,48 , 3070,14 , 7456,21 , 7477,48 , 3070,14 , 0,2 , 0,2 , {82,79,78},  0,0 , 5199,57 , 25,5 , 4,0 , 2253,6 , 2276,7 ,     2,     1,     1,     6,     7 }, // Romanian/AnyScript/Romania
@@ -562,7 +562,7 @@ static const QLocalePrivate locale_data[] = {
     {    106,     0,   198,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 572,10 , 959,17 , 522,7 , 25,12 , 12541,59 , 12600,96 , 12696,32 , 12808,61 , 12869,96 , 12965,32 , 8367,30 , 8397,62 , 8459,19 , 8367,30 , 8397,62 , 8459,19 , 293,5 , 276,4 , {76,75,82},  256,3 , 6172,19 , 4,4 , 13,6 , 2496,5 , 2501,11 ,     2,     1,     1,     6,     7 }, // Singhalese/AnyScript/SriLanka
     {    107,     0,   195,    44,   160,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 12728,48 , 12776,114 , 1381,27 , 12997,48 , 13045,114 , 134,27 , 8478,27 , 8505,68 , 708,14 , 8478,27 , 8505,68 , 708,14 , 0,2 , 0,2 , {90,65,82},  11,1 , 0,7 , 4,4 , 4,0 , 2512,7 , 0,0 ,     2,     1,     7,     6,     7 }, // Siswati/AnyScript/SouthAfrica
     {    107,     0,   204,    44,   160,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 12728,48 , 12776,114 , 1381,27 , 12997,48 , 13045,114 , 134,27 , 8478,27 , 8505,68 , 708,14 , 8478,27 , 8505,68 , 708,14 , 0,2 , 0,2 , {83,90,76},  259,1 , 0,7 , 4,4 , 4,0 , 2512,7 , 0,0 ,     2,     1,     1,     6,     7 }, // Siswati/AnyScript/Swaziland
-    {    108,     0,   191,  1643,  1644,  1563,  1642,  1632,    45,    43,   101,  8218,  8216,  8222,  8220, 0,6 , 0,6 , 104,7 , 104,7 , 582,8 , 498,18 , 165,4 , 195,9 , 12890,48 , 12938,82 , 12028,24 , 13159,48 , 13207,89 , 12295,24 , 8573,21 , 8594,52 , 8646,14 , 8573,21 , 8594,52 , 8646,14 , 298,10 , 280,9 , {69,85,82},  112,1 , 6191,28 , 25,5 , 4,0 , 2519,10 , 2529,19 ,     2,     1,     1,     6,     7 }, // Slovak/AnyScript/Slovakia
+    {    108,     0,   191,    44,   160,    44,    37,    48,    45,    43,   101,  8218,  8216,  8222,  8220, 0,6 , 0,6 , 104,7 , 104,7 , 582,8 , 498,18 , 165,4 , 195,9 , 12890,48 , 12938,82 , 12028,24 , 13159,48 , 13207,89 , 12295,24 , 8573,21 , 8594,52 , 8646,14 , 8573,21 , 8594,52 , 8646,14 , 298,10 , 280,9 , {69,85,82},  112,1 , 6191,28 , 25,5 , 4,0 , 2519,10 , 2529,19 ,     2,     1,     1,     6,     7 }, // Slovak/AnyScript/Slovakia
     {    109,     0,   192,    44,    46,    59,    37,    48,    45,    43,   101,   187,   171,  8222,  8220, 0,6 , 0,6 , 367,8 , 367,8 , 976,9 , 590,19 , 37,5 , 8,10 , 11899,48 , 13020,86 , 12028,24 , 13296,59 , 13355,86 , 12295,24 , 8660,28 , 8688,52 , 8740,14 , 8754,35 , 8688,52 , 8740,14 , 78,4 , 289,4 , {69,85,82},  112,1 , 6219,28 , 25,5 , 4,0 , 2548,11 , 2559,9 ,     2,     1,     1,     6,     7 }, // Slovenian/AnyScript/Slovenia
     {    110,     0,   194,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 27,8 , 53,19 , 18,7 , 25,12 , 13106,48 , 13154,189 , 13343,24 , 13441,48 , 13489,189 , 13678,24 , 8789,28 , 8817,47 , 8864,14 , 8789,28 , 8817,47 , 8864,14 , 308,3 , 293,3 , {83,79,83},  0,0 , 6247,22 , 4,4 , 4,0 , 2568,8 , 2576,10 ,     0,     0,     1,     6,     7 }, // Somali/AnyScript/Somalia
     {    110,     0,    59,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 27,8 , 53,19 , 18,7 , 25,12 , 13106,48 , 13154,189 , 13343,24 , 13441,48 , 13489,189 , 13678,24 , 8789,28 , 8817,47 , 8864,14 , 8789,28 , 8817,47 , 8864,14 , 308,3 , 293,3 , {68,74,70},  5,3 , 6269,21 , 4,4 , 4,0 , 2568,8 , 2586,7 ,     0,     0,     6,     6,     7 }, // Somali/AnyScript/Djibouti
@@ -597,9 +597,9 @@ static const QLocalePrivate locale_data[] = {
     {    115,     0,   170,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 408,8 , 408,8 , 554,6 , 1068,18 , 37,5 , 8,10 , 13747,48 , 13795,88 , 13883,24 , 14081,48 , 14129,88 , 14217,24 , 9162,28 , 9190,55 , 9245,14 , 9259,28 , 9190,55 , 9245,14 , 0,2 , 0,2 , {80,72,80},  139,1 , 7392,22 , 8,5 , 4,0 , 2896,7 , 2903,9 ,     2,     1,     7,     6,     7 }, // Tagalog/AnyScript/Philippines
     {    116,     0,   209,    46,    44,    59,    37,    48,    45,    43,   101,   171,   187,   171,  8222, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 13907,48 , 13955,71 , 1381,27 , 14241,48 , 14289,71 , 134,27 , 9287,28 , 9315,55 , 708,14 , 9287,28 , 9315,55 , 708,14 , 0,2 , 0,2 , {84,74,83},  186,3 , 7414,13 , 8,5 , 4,0 , 0,0 , 0,0 ,     2,     1,     1,     6,     7 }, // Tajik/AnyScript/Tajikistan
     {    116,     2,   209,    46,    44,    59,    37,    48,    45,    43,   101,   171,   187,   171,  8222, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 13907,48 , 13955,71 , 1381,27 , 14241,48 , 14289,71 , 134,27 , 9287,28 , 9315,55 , 708,14 , 9287,28 , 9315,55 , 708,14 , 0,2 , 0,2 , {84,74,83},  186,3 , 7414,13 , 8,5 , 4,0 , 0,0 , 0,0 ,     2,     1,     1,     6,     7 }, // Tajik/Cyrillic/Tajikistan
-    {    117,     0,   100,    46,    44,  1563,  1642,  3046,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 416,13 , 416,13 , 634,6 , 203,18 , 18,7 , 25,12 , 14026,58 , 14084,86 , 14170,31 , 14360,58 , 14418,86 , 14504,31 , 9370,20 , 9390,49 , 9370,20 , 9370,20 , 9390,49 , 9370,20 , 320,8 , 305,8 , {73,78,82},  114,1 , 7427,13 , 8,5 , 4,0 , 2912,5 , 2917,7 ,     2,     1,     7,     7,     7 }, // Tamil/AnyScript/India
-    {    117,     0,   198,    46,    44,  1563,  1642,  3046,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 416,13 , 416,13 , 634,6 , 203,18 , 18,7 , 25,12 , 14026,58 , 14084,86 , 14170,31 , 14360,58 , 14418,86 , 14504,31 , 9370,20 , 9390,49 , 9370,20 , 9370,20 , 9390,49 , 9370,20 , 320,8 , 305,8 , {76,75,82},  282,3 , 7440,20 , 8,5 , 4,0 , 2912,5 , 2924,6 ,     2,     1,     1,     6,     7 }, // Tamil/AnyScript/SriLanka
-    {    119,     0,   100,  1643,  1644,  1563,  1642,  3174,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 429,11 , 429,11 , 539,8 , 99,16 , 18,7 , 25,12 , 14201,78 , 14279,86 , 14365,30 , 14535,78 , 14613,86 , 14699,31 , 9439,32 , 9471,60 , 9531,18 , 9439,32 , 9471,60 , 9531,18 , 328,10 , 313,8 , {73,78,82},  285,3 , 7460,13 , 8,5 , 4,0 , 2930,6 , 2936,9 ,     2,     1,     7,     7,     7 }, // Telugu/AnyScript/India
+    {    117,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 416,13 , 416,13 , 634,6 , 203,18 , 18,7 , 25,12 , 14026,58 , 14084,86 , 14170,31 , 14360,58 , 14418,86 , 14504,31 , 9370,20 , 9390,49 , 9370,20 , 9370,20 , 9390,49 , 9370,20 , 320,8 , 305,8 , {73,78,82},  114,1 , 7427,13 , 8,5 , 4,0 , 2912,5 , 2917,7 ,     2,     1,     7,     7,     7 }, // Tamil/AnyScript/India
+    {    117,     0,   198,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 416,13 , 416,13 , 634,6 , 203,18 , 18,7 , 25,12 , 14026,58 , 14084,86 , 14170,31 , 14360,58 , 14418,86 , 14504,31 , 9370,20 , 9390,49 , 9370,20 , 9370,20 , 9390,49 , 9370,20 , 320,8 , 305,8 , {76,75,82},  282,3 , 7440,20 , 8,5 , 4,0 , 2912,5 , 2924,6 ,     2,     1,     1,     6,     7 }, // Tamil/AnyScript/SriLanka
+    {    119,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 429,11 , 429,11 , 539,8 , 99,16 , 18,7 , 25,12 , 14201,78 , 14279,86 , 14365,30 , 14535,78 , 14613,86 , 14699,31 , 9439,32 , 9471,60 , 9531,18 , 9439,32 , 9471,60 , 9531,18 , 328,10 , 313,8 , {73,78,82},  285,3 , 7460,13 , 8,5 , 4,0 , 2930,6 , 2936,9 ,     2,     1,     7,     7,     7 }, // Telugu/AnyScript/India
     {    120,     0,   211,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 440,5 , 440,5 , 445,8 , 453,7 , 364,8 , 1086,19 , 165,4 , 558,27 , 14395,63 , 14458,98 , 14395,63 , 14730,63 , 14793,98 , 14891,62 , 9549,23 , 9572,68 , 9640,14 , 9549,23 , 9572,68 , 9640,14 , 338,10 , 321,10 , {84,72,66},  288,1 , 7473,13 , 4,4 , 48,5 , 2945,3 , 2945,3 ,     2,     1,     7,     6,     7 }, // Thai/AnyScript/Thailand
     {    121,     0,    44,    46,    44,  1563,  1642,  3872,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 14556,63 , 14619,158 , 1381,27 , 14953,63 , 15016,158 , 134,27 , 9654,51 , 9705,79 , 9784,21 , 9805,49 , 9854,77 , 9931,22 , 348,7 , 331,8 , {67,78,89},  208,3 , 7486,13 , 8,5 , 4,0 , 2948,8 , 2956,6 ,     2,     1,     7,     6,     7 }, // Tibetan/AnyScript/China
     {    121,     0,   100,    46,    44,  1563,  1642,  3872,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 72,10 , 314,18 , 37,5 , 8,10 , 14556,63 , 14619,158 , 1381,27 , 14953,63 , 15016,158 , 134,27 , 9654,51 , 9705,79 , 9784,21 , 9805,49 , 9854,77 , 9931,22 , 348,7 , 331,8 , {73,78,82},  114,1 , 7499,22 , 8,5 , 4,0 , 2948,8 , 2962,7 ,     2,     1,     7,     7,     7 }, // Tibetan/AnyScript/India
@@ -612,7 +612,7 @@ static const QLocalePrivate locale_data[] = {
     {    130,     0,   100,  1643,  1644,  1563,  1642,  1776,    45,    43,   101,    34,    34,    39,    39, 31,6 , 31,6 , 476,10 , 486,9 , 279,6 , 1189,18 , 18,7 , 25,12 , 15523,67 , 15523,67 , 1381,27 , 15931,67 , 15931,67 , 134,27 , 10414,36 , 10414,36 , 708,14 , 10414,36 , 10414,36 , 708,14 , 364,2 , 348,3 , {73,78,82},  114,1 , 7588,18 , 8,5 , 4,0 , 3033,4 , 3037,5 ,     2,     1,     7,     7,     7 }, // Urdu/AnyScript/India
     {    130,     0,   163,    46,    44,    59,    37,    48,    45,    43,   101,    34,    34,    39,    39, 31,6 , 31,6 , 476,10 , 486,9 , 279,6 , 1189,18 , 18,7 , 25,12 , 15523,67 , 15523,67 , 1381,27 , 15931,67 , 15931,67 , 134,27 , 10414,36 , 10414,36 , 708,14 , 10414,36 , 10414,36 , 708,14 , 364,2 , 348,3 , {80,75,82},  291,4 , 7606,21 , 4,4 , 4,0 , 3033,4 , 3042,7 ,     0,     0,     7,     6,     7 }, // Urdu/AnyScript/Pakistan
     {    131,     0,   228,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 13907,48 , 15590,115 , 11500,24 , 14241,48 , 15998,115 , 11767,24 , 10450,28 , 10478,53 , 10531,14 , 10450,28 , 10478,53 , 10531,14 , 0,2 , 0,2 , {85,90,83},  295,3 , 7627,21 , 8,5 , 4,0 , 3049,5 , 3054,10 ,     0,     0,     1,     6,     7 }, // Uzbek/AnyScript/Uzbekistan
-    {    131,     0,     1,  1643,  1644,  1563,  1642,  1776,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 179,8 , 1207,33 , 165,4 , 452,11 , 15705,48 , 15753,68 , 1381,27 , 16113,48 , 10597,68 , 134,27 , 10545,21 , 6840,49 , 708,14 , 10545,21 , 6840,49 , 708,14 , 0,2 , 0,2 , {65,70,78},  215,1 , 7648,13 , 25,5 , 4,0 , 3064,6 , 2077,9 ,     0,     0,     6,     4,     5 }, // Uzbek/AnyScript/Afghanistan
+    {    131,     0,     1,    44,    46,    59,    37,    48,  8722,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 179,8 , 1207,33 , 165,4 , 452,11 , 15705,48 , 15753,68 , 1381,27 , 16113,48 , 10597,68 , 134,27 , 10545,21 , 6840,49 , 708,14 , 10545,21 , 6840,49 , 708,14 , 0,2 , 0,2 , {65,70,78},  215,1 , 7648,13 , 25,5 , 4,0 , 3064,6 , 2077,9 ,     0,     0,     6,     4,     5 }, // Uzbek/AnyScript/Afghanistan
     {    131,     1,     1,  1643,  1644,  1563,  1642,  1776,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 179,8 , 1207,33 , 165,4 , 452,11 , 15705,48 , 15753,68 , 1381,27 , 16113,48 , 10597,68 , 134,27 , 10545,21 , 6840,49 , 708,14 , 10545,21 , 6840,49 , 708,14 , 0,2 , 0,2 , {65,70,78},  215,1 , 7648,13 , 25,5 , 4,0 , 3064,6 , 2077,9 ,     0,     0,     6,     4,     5 }, // Uzbek/Arabic/Afghanistan
     {    131,     2,   228,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 13907,48 , 15590,115 , 11500,24 , 14241,48 , 15998,115 , 11767,24 , 10450,28 , 10478,53 , 10531,14 , 10450,28 , 10478,53 , 10531,14 , 0,2 , 0,2 , {85,90,83},  295,3 , 7627,21 , 8,5 , 4,0 , 3049,5 , 3054,10 ,     0,     0,     1,     6,     7 }, // Uzbek/Cyrillic/Uzbekistan
     {    131,     7,   228,    44,   160,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 15821,52 , 15873,77 , 15950,24 , 16161,52 , 16213,77 , 16290,24 , 10566,34 , 10600,61 , 10661,14 , 10566,34 , 10600,61 , 10661,14 , 0,2 , 0,2 , {85,90,83},  298,4 , 7661,23 , 8,5 , 4,0 , 3070,9 , 3079,11 ,     0,     0,     1,     6,     7 }, // Uzbek/Latin/Uzbekistan
@@ -626,7 +626,7 @@ static const QLocalePrivate locale_data[] = {
     {    144,     0,   224,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 27,8 , 82,17 , 37,5 , 8,10 , 16911,102 , 17013,140 , 1381,27 , 17245,102 , 17347,140 , 134,27 , 11353,30 , 11383,57 , 708,14 , 11353,30 , 11383,57 , 708,14 , 61,4 , 59,4 , {71,66,80},  140,1 , 0,7 , 4,4 , 4,0 , 3207,5 , 3212,14 ,     2,     1,     1,     6,     7 }, // Manx/AnyScript/UnitedKingdom
     {    145,     0,   224,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 99,16 , 37,5 , 8,10 , 17153,46 , 17199,124 , 1381,27 , 17487,46 , 17533,124 , 134,27 , 11440,28 , 11468,60 , 708,14 , 11440,28 , 11468,60 , 708,14 , 61,4 , 59,4 , {71,66,80},  140,1 , 0,7 , 4,4 , 4,0 , 3226,8 , 3212,14 ,     2,     1,     1,     6,     7 }, // Cornish/AnyScript/UnitedKingdom
     {    146,     0,    83,    46,   160,    59,    37,    48,    45,    43,   101,  8216,  8217,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 17323,48 , 17371,192 , 1381,27 , 17657,48 , 17705,192 , 134,27 , 11528,28 , 11556,49 , 11605,14 , 11528,28 , 11556,49 , 11605,14 , 382,2 , 369,2 , {71,72,83},  0,0 , 0,7 , 4,4 , 4,0 , 3234,4 , 3238,5 ,     2,     1,     1,     6,     7 }, // Akan/AnyScript/Ghana
-    {    147,     0,   100,  1643,  1644,  1563,  1642,  2406,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 634,6 , 99,16 , 18,7 , 25,12 , 17563,87 , 17563,87 , 1381,27 , 17897,87 , 17897,87 , 134,27 , 6311,32 , 11619,55 , 708,14 , 6311,32 , 11619,55 , 708,14 , 218,5 , 201,5 , {73,78,82},  205,2 , 0,7 , 8,5 , 4,0 , 3243,6 , 1640,4 ,     2,     1,     7,     7,     7 }, // Konkani/AnyScript/India
+    {    147,     0,   100,    46,    44,    59,    37,    48,    45,    43,   101,  8220,  8221,  8216,  8217, 0,6 , 0,6 , 0,6 , 0,6 , 634,6 , 99,16 , 18,7 , 25,12 , 17563,87 , 17563,87 , 1381,27 , 17897,87 , 17897,87 , 134,27 , 6311,32 , 11619,55 , 708,14 , 6311,32 , 11619,55 , 708,14 , 218,5 , 201,5 , {73,78,82},  205,2 , 0,7 , 8,5 , 4,0 , 3243,6 , 1640,4 ,     2,     1,     7,     7,     7 }, // Konkani/AnyScript/India
     {    148,     0,    83,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,    34,    34, 0,6 , 0,6 , 0,6 , 0,6 , 221,8 , 314,18 , 37,5 , 8,10 , 17650,48 , 17698,94 , 1381,27 , 17984,48 , 18032,94 , 134,27 , 11674,26 , 11700,34 , 708,14 , 11674,26 , 11700,34 , 708,14 , 0,2 , 0,2 , {71,72,83},  0,0 , 0,7 , 8,5 , 4,0 , 0,0 , 0,0 ,     2,     1,     1,     6,     7 }, // Ga/AnyScript/Ghana
     {    149,     0,   157,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 10,17 , 18,7 , 25,12 , 17792,48 , 17840,86 , 1381,27 , 18126,48 , 18174,86 , 134,27 , 11734,29 , 11763,57 , 708,14 , 11734,29 , 11763,57 , 708,14 , 384,4 , 371,4 , {78,71,78},  171,1 , 7826,12 , 4,4 , 13,6 , 3249,4 , 3253,7 ,     2,     1,     1,     6,     7 }, // Igbo/AnyScript/Nigeria
     {    150,     0,   111,    46,    44,    59,    37,    48,    45,    43,   101,    39,    39,  8220,  8221, 0,6 , 0,6 , 0,6 , 0,6 , 141,10 , 10,17 , 18,7 , 25,12 , 17926,48 , 17974,189 , 18163,24 , 18260,48 , 18308,189 , 18497,24 , 11820,28 , 11848,74 , 11922,14 , 11820,28 , 11848,74 , 11922,14 , 388,9 , 375,7 , {75,69,83},  2,3 , 7838,23 , 4,4 , 13,6 , 3260,7 , 2862,5 ,     2,     1,     7,     6,     7 }, // Kamba/AnyScript/Kenya
-- 
1.8.1.4

